packages/studio/src/store/models/slices/PagePreviewSlice.ts

```diff
@@ -0,0 +1,6 @@
+import { ViewportStyles } from "../../../components/Viewport/defaults";
+
+export default interface PagePreviewSlice {
+  viewportDimensions: ViewportStyles;
```

Comment:
What I'm thinking is we have this:

```
viewport: Viewport;
setViewport(viewport: Viewport);
```

In the drop-down you've created, we'd modify `handleSelect` to look like this:

```
  const handleSelect = useCallback(() => {
    closeMenu();
    const viewport = VIEWPORT.values().findFirst(viewport.name === name);
    // Error handling to make sure `viewport` exists.
    // Apply default styles to viewport if `styles.height` and `styles.width` are missing
    setViewport(viewport);
  }, [closeMenu, name, setViewportDimensions, styles]);
```

In my mind, the main benefit to this approach is that we don't need to look in two different places for information about the Viewport. Right now, in `IFramePortal`, we get the dimensions of the current Viewport from global state, but the CSS from the static map. It'd be nice if we could get the entire, current `Viewport` from State. If this doesn't work because of Tailwind dynamic class name issues, lemme know!